exporter/datasetexporter/config_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datasetexporter
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/maxatome/go-testdeep/helpers/tdsuite"
+	"github.com/maxatome/go-testdeep/td"
```

Comment:
The guidance for testing libraries suggests using [stretchr/testify](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#testing-library-recommendations) for anything needed beyond the Go standard library for suites/assertions/etc. Ideally we'd be consistent with this across the Collector, but the advice in the Collector core isn't a hard requirement. Is there an advantage to using this library over testify?

Reply:
It was used by github.com/jarcoal/httpmock that I am using in the tests for the library - https://github.com/scalyr/dataset-go - that will be used by this exporter. I can swap it.